community/kernel/src/main/java/org/neo4j/kernel/impl/api/ConstraintEnforcingEntityOperations.java

```diff
@@ -614,23 +638,27 @@ public MandatoryNodePropertyConstraint mandatoryNodePropertyConstraintCreate( Ke
         PrimitiveLongIterator nodes = nodesGetForLabel( state, labelId );
         while ( nodes.hasNext() )
         {
-            try
+            long nodeId = nodes.next();
+            try ( Cursor<NodeItem> node = nodeCursor( state, nodeId ) )
             {
-                long nodeId = nodes.next();
-                if ( !nodeHasProperty( state, nodeId, propertyKeyId ) )
+                if ( !node.next() )
                 {
-                    MandatoryNodePropertyConstraint constraint = new MandatoryNodePropertyConstraint( labelId, propertyKeyId );
+                    PropertyConstraint constraint = new MandatoryNodePropertyConstraint( labelId, propertyKeyId );
                     throw new CreateConstraintFailureException( constraint,
-                            new MandatoryNodePropertyConstraintVerificationFailedKernelException( constraint, nodeId ) );
+                            new EntityNotFoundException( EntityType.NODE, nodeId ) );
```

Comment:
You're right, if we can't find the node the right thing to do is to just skip it.